atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/SchemaMutationLockTables.java

```diff
@@ -63,31 +67,59 @@ public SchemaMutationLockTables(CassandraClientPool clientPool, CassandraKeyValu
     }
 
     public TableReference createLockTable() throws TException {
-        return clientPool.runWithRetry(this::createLockTable);
-    }
+        TableReference lockTable = TableReference.createWithEmptyNamespace(LOCK_TABLE_PREFIX);
 
-    private TableReference createLockTable(CassandraClient client) throws TException {
-        String lockTableName = LOCK_TABLE_PREFIX + "_" + getUniqueSuffix();
-        TableReference lockTable = TableReference.createWithEmptyNamespace(lockTableName);
-        log.info("Creating lock table {}", SafeArg.of("schemaMutationTableName", lockTable));
-        createTableInternal(client, lockTable);
-        return lockTable;
+        try {
+            log.info("Creating lock table {}", SafeArg.of("schemaMutationTableName", lockTable.getQualifiedName()));
+            createTableWithCustomId(lockTable);
+            return lockTable;
+        } catch (InvalidRequestException ire) {
+            // This can happen if multiple nodes concurrently attempt to create the locks table
+            Set<TableReference> lockTables = getAllLockTables();
+            if (lockTables.size() == 1) {
+                log.info("Lock table was created by another node - returning it.");
+                return Iterables.getOnlyElement(lockTables);
+            } else {
+                throw new IllegalStateException(String.format("Expected 1 locks table to exist, but found %d",
+                        lockTables.size()));
+            }
+        }
     }
 
-    private String getUniqueSuffix() {
-        // We replace '-' with '_' as hyphens are forbidden in Cassandra table names.
-        return UUID.randomUUID().toString().replace('-', '_');
-    }
+    private void createTableWithCustomId(TableReference tableRef) throws TException {
+        String internalTableName = CassandraKeyValueServiceImpl.internalTableName(tableRef);
+        String keyspace = config.getKeyspaceOrThrow();
+        String fullTableNameForUuid = keyspace + "." + internalTableName;
+        UUID uuid = UUID.nameUUIDFromBytes(fullTableNameForUuid.getBytes());
+
+        String createTableStatement = "CREATE TABLE \"%s\".\"%s\" (\n"
```

Comment:
I'm not sure how much of this are just cassandra defaults (or even our own internal defaults) but the full schema representation I see in a production stack looks like:
```
CREATE TABLE gatekeeper."_locks_706c8369_7883_4add_974c_882172a005fb" (
    key blob,
    column1 blob,
    column2 bigint,
    value blob,
    PRIMARY KEY (key, column1, column2)
) WITH COMPACT STORAGE
    AND CLUSTERING ORDER BY (column1 ASC, column2 ASC)
    AND bloom_filter_fp_chance = 0.1
    AND caching = '{"keys":"ALL", "rows_per_partition":"NONE"}'
    AND comment = ''
    AND compaction = {'class': 'org.apache.cassandra.db.compaction.LeveledCompactionStrategy'}
    AND compression = {}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 3600
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = 'NONE';
```

Reply:
So, that's precisely what I had before commit [acb6a36](https://github.com/palantir/atlasdb/pull/3088/commits/acb6a3662aec8d13f3cf05c755250b272b8c9c12).

You said "I suggest making this simpler by having the method be very schema mutation lock table specific so you don't have to worry about arbitrary column family definitions", which I took to mean "get rid of parameters we set for regular tables but don't need to tiny locks table" - maybe I was overzealous/mistaken?

Much of the values are set in [ColumnFamilyDefinitions](https://github.com/palantir/atlasdb/blob/0.80.0/atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitions.java#L133).